sw/device/sca/aes_serial.c

```diff
@@ -198,18 +203,68 @@ static void init_aes(void) {
  * UART.
  */
 int main(void) {
-  const dif_uart_t *uart;
+  const dif_uart_t *uart1;
 
   sca_init();
-  sca_get_uart(&uart);
+  sca_get_uart(&uart1);
 
-  simple_serial_init(uart);
+  LOG_INFO("Running AES serial");
+
+  LOG_INFO("Disabling entropy complex and unneeded clocks to reduce noise.");
+  // Disable the entropy complex to reduce the noise in the power traces. We can
+  // only do this as AES features a parameter to skip PRNG reseeding for SCA
+  // experiments. Without this parameter, AES would simply get stalled with a
+  // disabled entropy complex.
+
+  // Disable/stopping functionality not yet provided by EDN and CSRNG DIFs.
+  mmio_region_write32(mmio_region_from_addr(TOP_EARLGREY_EDN0_BASE_ADDR),
+                      EDN_CTRL_REG_OFFSET, EDN_CTRL_REG_RESVAL);
+  mmio_region_write32(mmio_region_from_addr(TOP_EARLGREY_EDN1_BASE_ADDR),
+                      EDN_CTRL_REG_OFFSET, EDN_CTRL_REG_RESVAL);
+  mmio_region_write32(mmio_region_from_addr(TOP_EARLGREY_CSRNG_BASE_ADDR),
+                      CSRNG_CTRL_REG_OFFSET, CSRNG_CTRL_REG_RESVAL);
+
+  // Disable entropy source through DIF.
+  const dif_entropy_params_t entropy_params = {
+      .base_addr = mmio_region_from_addr(TOP_EARLGREY_ENTROPY_SRC_BASE_ADDR),
+  };
+  dif_entropy_t entropy;
+  CHECK(dif_entropy_init(entropy_params, &entropy) == kDifEntropyOk);
+  CHECK(dif_entropy_disable(&entropy) == kDifEntropyOk);
+
+  // Disable HMAC, KMAC, OTBN and USB clocks through CLKMGR DIF.
+  const dif_clkmgr_params_t clkmgr_params = {
+      .base_addr = mmio_region_from_addr(TOP_EARLGREY_CLKMGR_AON_BASE_ADDR),
+      .last_gateable_clock = CLKMGR_CLK_ENABLES_CLK_USB_PERI_EN_BIT,
+      .last_hintable_clock = CLKMGR_CLK_HINTS_STATUS_CLK_MAIN_OTBN_VAL_BIT};
+  dif_clkmgr_t clkmgr;
+  CHECK(dif_clkmgr_init(clkmgr_params, &clkmgr) == kDifClkmgrOk);
+  CHECK(dif_clkmgr_hintable_clock_set_hint(
+            &clkmgr, CLKMGR_CLK_HINTS_CLK_MAIN_HMAC_HINT_BIT,
+            kDifClkmgrToggleDisabled) == kDifClkmgrOk);
+  CHECK(dif_clkmgr_hintable_clock_set_hint(
+            &clkmgr, CLKMGR_CLK_HINTS_CLK_MAIN_KMAC_HINT_BIT,
+            kDifClkmgrToggleDisabled) == kDifClkmgrOk);
+  CHECK(dif_clkmgr_hintable_clock_set_hint(
+            &clkmgr, CLKMGR_CLK_HINTS_CLK_IO_DIV4_OTBN_HINT_BIT,
+            kDifClkmgrToggleDisabled) == kDifClkmgrOk);
+  CHECK(dif_clkmgr_hintable_clock_set_hint(
+            &clkmgr, CLKMGR_CLK_HINTS_CLK_MAIN_OTBN_HINT_BIT,
+            kDifClkmgrToggleDisabled) == kDifClkmgrOk);
+  CHECK(dif_clkmgr_gateable_clock_set_enabled(
+            &clkmgr, CLKMGR_CLK_ENABLES_CLK_USB_PERI_EN_BIT,
+            kDifClkmgrToggleDisabled) == kDifClkmgrOk);
```

Comment:
I've switched to `IGNORE_RESULT` and moved this part to `sca.c` as suggested. What is missing are function arguments to specify which clocks to disable etc. This will have to be introduced once we perform SCA for other IPs (as other parts of `sca.c` which is still a bit AES focused).